etl/steps/data/garden/cedlas/2024-03-08/sedlac.py

```diff
@@ -0,0 +1,176 @@
+}
+
+INEQUALITY_GINI_TABLES = {
+    "gini1": {"column_name": "gini"},
+    "gini2": {"column_name": "gini"},
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("sedlac")
+
+    # Read tables from meadow dataset.
+    tb_poverty = load_tables(dataset=ds_meadow, tables_dict=POVERTY_TABLES)
+    tb_ineq_indices = load_tables(dataset=ds_meadow, tables_dict=INEQUALITY_INDICES_TABLES)
+    tb_ineq_deciles = load_tables(dataset=ds_meadow, tables_dict=INEQUALITY_DECILES_TABLES)
+
+    #
+    # Process data.
+    tb_poverty = merge_tables(tables=tb_poverty, merge_or_concat="concat", short_name="poverty")
+    tb_ineq_indices = merge_tables(tables=tb_ineq_indices, merge_or_concat="concat", short_name="ineq_indices")
+    tb_ineq_deciles = merge_tables(tables=tb_ineq_deciles, merge_or_concat="concat", short_name="ineq_deciles")
+
+    tb_inequality = merge_tables(
+        tables=[tb_ineq_indices, tb_ineq_deciles], merge_or_concat="merge", short_name="inequality"
+    )
+
+    tb = merge_tables(tables=[tb_poverty, tb_inequality], merge_or_concat="merge", short_name="sedlac")
+
+    tb = geo.harmonize_countries(
+        df=tb,
+        countries_file=paths.country_mapping_path,
+    )
+
+    # Remove urban and rural estimates from poverty table
+    tb = drop_urban_rural_from_poverty_table(tb)
+
+    # Remove duplicate rows and spells from tables, for Grapher datasets
+    tb_no_spells = remove_spells_duplicates_and_add_urban(tb=tb, short_name="sedlac_no_spells")
+
+    tb = tb.set_index(["country", "year", "survey_number", "survey", "table"], verify_integrity=True)
+    tb_no_spells = tb_no_spells.set_index(["country", "year", "table"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir,
+        tables=[tb, tb_no_spells],
+        check_variables_metadata=True,
+        default_metadata=ds_meadow.metadata,
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def load_tables(dataset: Dataset, tables_dict: Dict) -> List[Table]:
+    """Load tables from the meadow dataset."""
+
+    tables = []
+    for t in tables_dict:
+        tb = dataset[t].reset_index()
+
+        # Create a column with the name of the table
+        tb["table"] = tables_dict[t]["column_name"]
+
+        tables.append(tb)
+
+    return tables
+
+
+def merge_tables(tables: List[Table], merge_or_concat: str, short_name: str) -> Table:
+    """Merge tables into a single table."""
+
+    if merge_or_concat == "concat":
+        # Concatenate tables
+        tb = pr.concat(tables, ignore_index=True, short_name=short_name)
+    elif merge_or_concat == "merge":
+        # use the first table as base
+        tb = tables[0]
+        for i in range(len(tables) - 1):
+            if i < len(tables) - 1:
```

Comment:
don't think you need this if statement since you are already iterating only n = (tables-1)

Reply:
That's right, there was a moment where I was merging more tables. I am waiting for Joe if he wants those tables (probably not)